src/main/java/net/obvj/performetrics/util/Duration.java

```diff
@@ -424,30 +227,22 @@ public double toTimeUnit(TimeUnit timeUnit)
      */
     public double toTimeUnit(TimeUnit timeUnit, int scale)
     {
-        BigDecimal targetHours = hours > 0
-                ? BigDecimal.valueOf(timeUnit.convert(hours, TimeUnit.HOURS))
-                : BigDecimal.ZERO;
-
-        BigDecimal targetMinutes = minutes > 0
-                ? BigDecimal.valueOf(timeUnit.convert(minutes, TimeUnit.MINUTES))
-                : BigDecimal.ZERO;
-
-        BigDecimal targetSeconds = seconds > 0
-                ? BigDecimal.valueOf(timeUnit.convert(seconds, TimeUnit.SECONDS))
+        BigDecimal targetSeconds = internalDuration.getSeconds() > 0
+                ? BigDecimal.valueOf(timeUnit.convert(internalDuration.getSeconds(), TimeUnit.SECONDS))
                 : BigDecimal.ZERO;
 
-        BigDecimal targetNanoseconds = nanoseconds > 0
+        BigDecimal targetNanoseconds = internalDuration.getNano() > 0
                 ? convertNanosecondsPart(timeUnit, scale)
                 : BigDecimal.ZERO;
 
-        return targetHours.add(targetMinutes).add(targetSeconds).add(targetNanoseconds).doubleValue();
+        return targetSeconds.add(targetNanoseconds).doubleValue();
     }
 
     private BigDecimal convertNanosecondsPart(TimeUnit timeUnit, int scale)
     {
         return scale >= 0
-                ? BigDecimal.valueOf(TimeUnitConverter.convertAndRound(nanoseconds, TimeUnit.NANOSECONDS, timeUnit, scale))
-                : BigDecimal.valueOf(TimeUnitConverter.convertAndRound(nanoseconds, TimeUnit.NANOSECONDS, timeUnit));
+                ? BigDecimal.valueOf(TimeUnitConverter.convertAndRound(internalDuration.getNano(), TimeUnit.NANOSECONDS, timeUnit, scale))
```

Comment:
Extract the content of `internalDuration.getNanos()` to a local variable to avoid duplicating the expression.